### DISCUSSION TITLE
Architecure Suggestions for Syslog Servers

### DISCUSSION THREAD

**Discussion by gebhard73** [Security-Onion-Solutions/securityonion#2947] [Unsupported Versions] [answered]
Hi,

we've some 16.04 boxes just for collecting and searching syslogs. The systems are completely separated for several reasons (data privacy, operations teams, ...) and operate independent of each other. This is meant for operation support (with a benefit for security) and works fine for us as of now.

As far as I know it is not possible to have something like access control for events: events from system A can only be seen by users U, V, events from system B can only be seen by users W, X, and so on. Please correct me if I'm wrong.

With 2.x we therfor want to start with the same set-up for syslog only as we have in place now: independent boxes for different teams, very different log volumes. What would be the suggested setup method? Standalone? Heavy Node? (Eval?)

Volume estimation:
- syslog from 6 boxes, about 500 GB per day, stored for about 24 hours
- syslog from 100 boxes, very low volume, stored as long as space lasts
- new system, to be build: syslog from about 10 boxes, about 1 TB per day, stored for about 14 days (if possible, further testing and PoC needed)

As I said this is only a start - we want to get used to SO so that we can extend to other features besides syslog in the long run for a known system when SO has proven its value to respective managers (going from operational support more into security monitoring).

Thanks for any input.

Regards,
Gebhard

**Comment by TOoSmOotH** [comment, marked as answer]
You would need to use standalone if you want to restrict users from seeing things in each of the instances.

**Reply by gebhard73** [reply]
Tnx. Would this be a candidate for a FR or is this a niche situation?

**Reply by ganduulgag** [reply]
@gebhard73 what is the resource  (CPU, RAM etc) allocation for the SO box you assign for each of these log setup? how did you estimate?

**Reply by gebhard73** [reply]
Hi,

we've already some syslog-only boxes with SO 16.04 so this was my baseline. Regarding sizing I'll have to check and will let you know.

Cheers.

**Reply by ganduulgag** [reply]
@gebhard73 thanks in advance.

**Reply by gebhard73** [reply]
So here's what we have right now for big boxes:

SO 2:
- max about 12 Mio events per hour
- 8 CPU
- 32 GB RAM
- retention time: 14 days (have not yes reached this limit)
- on ESX host, works fast

SO 16:
- max about 10 Mio events per hour
- 4 CPU
- 16 GB RAM
- retention time: 1 day
- on ESX host, works OK

Both host use fast dissect parsing (grok things have been disabled). See #3175 for some ideas for SO 2.

**Reply by ganduulgag** [reply]
@gebhard73 thanks. how did you parse the incoming syslog from other hosts?

**Reply by gebhard73** [reply]
Syslog port is working out of the box on SO 2 (it's forwarded to the filebeat container). The parsing happens in the ingest pipelines of elasticsearch. We use only the syslog file and distinguish the different hosts by e.g. hostname, IP range, other stuff. We then assign tags to each message for easy filtering.
